CONTRIBUTING.md

```diff
@@ -0,0 +1,249 @@
+* Search to see if your issue already exists or has even been solved previously.
+* If you indeed have a new issue or request, click the "New Issue" button.
+* Fill in as much of the issue template as is relevant. Please be as specific as possible. Include the version of the code you were using, as well as what operating system you are running. If possible, include complete, minimal example code that reproduces the problem.
+
+## Setting up your development environment
+
+We recommend using the [conda](https://conda.io/docs/) package manager for your Python environments.
+
+Install [git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+(link with instructions) on your system if not already available
+(check with ``git --version`` at the command line.)
+This can also be installed from a variety of package managers, including ``conda`` if needed.
+
+Login to your [GitHub](https://github.com) account and make a fork of the
+[IOOS_QC repository](https://github.com/ioos/ioos_qc) by clicking the "Fork" button.
+Clone your fork of the repository (in terminal on Mac/Linux or git shell/GUI on Windows)
+to the location you'd like to keep it.
+We are partial to creating a ``git_repos`` or ``projects`` directory in our home folder.
+
+```sh
+git clone https://github.com/<your-user-name>/ioos_qc.git
+```
+
+Navigate to that folder in the terminal or in Anaconda Prompt if you're on Windows.
+The remainder of the instructions will take place within this directory.
+
+```sh
+cd ioos_qc
+```
+
+Connect your repository to the upstream (main project).
+
+```sh
+git remote add upstream https://github.com/ioos/ioos_qc.git
+```
+
+Create a new conda environment for us to configure, and give it a name.
+
+For Environment Setup:
+
+After ``-n`` you can specify any name you'd like; here we've chosen ``ioosqc38``.
+
+```sh
+conda create -n ioosqc38 python=3.8
+```
+
+**IMPORTANT**: Always activate this environment when developing and testing your changes!
+
+```sh
+conda activate ioosqc38
+```
+
+You will have to do this any time you re-open your prompt.
+Currently there are no packages in this environment, let's change that.
+
+```sh
+conda install --file requirements.txt --file requirements-dev.txt
+```
+
+Note that those requirement file names may change depending on the repository.
+Again, always check the local README for more information.
+
+Finally, to run all the tests!
+
+```sh
+pytest
+```
+
+Once you are done with environemtn setup, its time to build docs:
+
+Create the following environment:
+
+```sh
+conda create -y -n ioosqc38_docs python=3.8
+```
+
+**IMPORTANT**: Always activate this environment when developing and testing your changes!
+
+```sh
+conda activate ioosqc38_docs
+```
+
+You will have to do this any time you re-open your prompt.
+Currently there are no packages in this environment, let's change that.
+
+```sh
+conda install -y --file requirements.txt \ --file requirements-dev.txt \ --file docs/requirements.txt
```

Comment:
We should simplify this later but it is OK for now. We don't need so many requirements file. We should probably add those to pyproject.toml.

```suggestion
conda install  --file requirements.txt --file requirements-dev.txt --file docs/requirements.txt
```